src/PartitionService.ts

```diff
@@ -106,7 +107,18 @@ export class PartitionServiceImpl implements PartitionService {
         if (typeof key === 'object' && 'getPartitionHash' in key) {
             partitionHash = key.getPartitionHash();
         } else {
-            partitionHash = this.serializationService.toData(key).getPartitionHash();
+            try {
```

Comment:
If key is not Data, we need to serialize it first to be able to get the partition hash of it. While doing so, a SchemaReplicatedError might be thrown if the key is a compact serializable class and the schema is not replicated to the cluster. In that case I rethrow the error to the user with a public error (HazelcastSerializationError) instead of an internal one, and enrich the message.

Java does not have it because this is problem is the problem of async clients.

Reply:
OK I guess what we don't have in .NET is the partition service throw in this case, because we manage schemas differently - so no issue here